### PR TITLE
fix(core): Fix new partial execution flow not working with workflows with multiple disabled nodes in a row (no-changelog)

### PR DESCRIPTION
**Pull request by despairblue**
## Summary

The implementation of `findSubgraph` was only able to remove a disabled node if it wasn't preceded by another disabled node. This fixes it. 

Before: 

https://github.com/user-attachments/assets/3cb57a55-d661-43fc-93f0-ea4d3a31b495

After:

https://github.com/user-attachments/assets/e31fea17-6936-4a75-ad3c-1d1c0ee941f5

Also I removed some old occurences of a function name that does not exist anymore.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/PAY-2035/partial-executions-not-working-when-there-are-2-disabled-nodes-in-a

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] ~[Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.~
- [x] Tests included. 
- [ ] ~PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)~

### CODE REVIEW

File: packages/core/src/PartialExecutionUtils/DirectedGraph.ts [hunk 1]
```diff
@@ -67,6 +67,63 @@ export class DirectedGraph {
...
+
+			const newConnections: GraphConnection[] = [];
+
+			for (const incomingConnection of incomingConnections) {
+				for (const outgoingConnection of outgoingConnections) {
+					const newConnection = {
+						...incomingConnection,
+						to: outgoingConnection.to,
+						inputIndex: outgoingConnection.inputIndex,
+					};
```
Comment (tomi) [inline packages/core/src/PartialExecutionUtils/DirectedGraph.ts:95]: Does this work if the out connection has fewer inputs than what the deleted node has? E.g. a case like this

```
		//                xx
		//  ┌─────┐     ┌─────┐     ┌─────┐
		//  │     │o   o│     │o ┌►o│node2│
		//  │     │o─┐ o│     │o │  └─────┘
		//  │node0│o └►o│node1│o │  
		//  │     │o   o│     │o─┤
		//  │     │o   o│     │o 
		//  └─────┘     └─────┘     
```

I guess it should cause if I understood correctly every incoming is connected to every output?

Comment (despairblue) [reply packages/core/src/PartialExecutionUtils/DirectedGraph.ts:95]: Good question. Let's try it out. I'll commit a test case for that.

Comment (despairblue) [reply packages/core/src/PartialExecutionUtils/DirectedGraph.ts:95]: See [`c70b710` (#11024)](https://github.com/n8n-io/n8n/pull/11024/commits/c70b7100e9ccf241f6d844b9042b149c343cb90f)

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.62.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.62.1)

**Pull request merged**